File: WebApplication_lab/Classes/Column.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace WebApplication.Classes
{
    public abstract class Column
    {
        public string Name { get; set; }
        public string Type { get; set; }

        protected Column(string name)
        {
            Name = name;
        }

        public abstract bool Validate(string data);
    }
}

File: WebServices_lab/src/main/java/org/example/component/Column.java
package org.example.component;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import org.example.parser.ColumnDeserializer;

import java.io.Serializable;

@JsonDeserialize(using = ColumnDeserializer.class)
public abstract class Column  implements Serializable {
    @JsonProperty("Name")
    public String name;
    @JsonProperty("Type")
    public String type;

    public Column() {
    }

    public Column(String name){
        this.name = name;
    }

    public String getName() {
        return name;
    }

    public String getType() {
        return type;
    }

    public void setName(String name) {
        this.name = name;
    }

    public abstract boolean validate(String data);
}
